### ISSUE TITLE
use redirect with base prefix

### ISSUE THREAD

**Issue by amoghh** [single-spa/single-spa-layout#156]
How do i get Single Spa Router work with base and redirects
the following is my router config
```
<template id="single-spa-layout">
    <single-spa-router base="/dev">
      <nav>
        <application name="navbar"></application>
      </nav>
      <main>
        <redirect from="/" to="/profile"></redirect>
        <route path="/profile">
          <application name="guest-spa"></application>
        </route>
        <route path="/offers">
          <application name="booking-spa"></application>
        </route>
      </main>
    </single-spa-router>
  </template>
```
My root config is hosted at xyz.com/dev/
i want path matching to happen with /dev as prefix hence i have used the base attribute but i also want redirects to happen from
/dev to /dev/profile and /dev/ to /dev/profile which isnt happening. please can anyone help

**Comment by amoghh** [comment]
If i set up the routing in the following manner (using the prefix '/dev' in the redirects manually)

```
<template id="single-spa-layout">
    <single-spa-router base="/dev">
      <nav>
        <application name="navbar"></application>
      </nav>
      <main>
        <redirect from="/dev" to="/profile"></redirect>
        <route path="/profile">
          <application name="guest-spa"></application>
        </route>
        <route path="/offers">
          <application name="booking-spa"></application>
        </route>
      </main>
    </single-spa-router>
  </template>
```
The path matching works fine and the redirect happens but only from `/dev to /dev/profile` The redirect does not work for `/dev/ to /dev/profile`

can anyone please help its a bit urgent

**Comment by joeldenning** [comment]
I agree that redirects should take into account the base url, although it will need to be a breaking change of single-spa-layout since there are likely those relying on the existing behavior.

> can anyone please help its a bit urgent

Trying to pressure myself and other maintainers to do work for you is not helpful, since you're using a free, open source project that you have not paid for. Please consider contributing a pull request implementing the feature rather than pressuring others to do work for you. I may be able to work on this later this week, but generally deprioritize issues where people pressure me or feel entitled to my time.

**Comment by amoghh** [comment]
Hey @joeldenning  sorry for coming of like that, i didn't mean to request a urgent change request or a bug fix, since there is lack of support on this library i was just confused whether i was doing something wrong hence i wanted to know whether there was a work around possible or if there was something that i was missing.

I necessarily don't require the redirect to work with base prefix, i can work around that by setting the full redirects (including prefixes) no issues but as i mentioned in my last comment i did manage to get it partially working but its not redirecting if there is a trailing slash in the URL. Is it something that i am missing or doing wrong?

suppose my application URL is `https:/google.com/dev`
if i enter `https:/google.com/dev` it successfully redirects to `https:/google.com/dev/profile` and my spa is loaded fine
but if i enter `https:/google.com/dev/` (with the slash) the redirect doesn't happen and my spa isn't loaded

The following is my template
```
<template id="single-spa-layout">
    <single-spa-router base="/dev">
      <nav>
        <application name="navbar"></application>
      </nav>
      <main>
        <redirect from="/dev" to="/dev/profile"></redirect>
        <route path="/profile">
          <application name="profile-spa"></application>
        </route>
      </main>
    </single-spa-router>
  </template>
```
